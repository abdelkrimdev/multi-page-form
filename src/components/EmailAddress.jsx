import React from 'react'
import PropTypes from 'prop-types'
import { Input, Icon } from 'antd'

const EmailAddress = ({ onChange }) => {
  return (
    <Input
      type='email'
      name='emailAddress'
      placeholder='peter.parker@marvel.co'
      prefix={<Icon type='mail' style={{ color: 'rgba(0,0,0,.25)' }} />}
      onChange={onChange}
      size='large'
    />
  )
}

EmailAddress.propTypes = {
  onChange: PropTypes.func
}

export default EmailAddress
