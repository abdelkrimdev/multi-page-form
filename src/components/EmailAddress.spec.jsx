import React from 'react'
import { shallow } from 'enzyme'

import EmailAddress from './EmailAddress'

describe('email address component', () => {
  it('should render without crashing.', () => {
    const wrapper = shallow(<EmailAddress />)

    expect(wrapper).toBeDefined()
  })

  it('should render correctly.', () => {
    const wrapper = shallow(<EmailAddress />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should set the onChange property.', () => {
    const handleChange = jest.fn()
    const wrapper = shallow(<EmailAddress onChange={handleChange} />)
    
    wrapper.prop('onChange')()

    expect(handleChange).toBeCalled()
  })
})
