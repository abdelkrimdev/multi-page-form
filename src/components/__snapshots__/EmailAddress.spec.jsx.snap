// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`email address component should render correctly. 1`] = `
<Input
  disabled={false}
  name="emailAddress"
  placeholder="peter.parker@marvel.co"
  prefix={
    <Icon
      style={
        Object {
          "color": "rgba(0,0,0,.25)",
        }
      }
      type="mail"
    />
  }
  prefixCls="ant-input"
  size="large"
  type="email"
/>
`;
